sale_order_lines_discount/sale_order_lines.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+from openerp.osv import osv
+from openerp.tools.translate import _
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+    _name = 'sale.order'
+
+    all_discounts = fields.Float(
+        'All order lines discount',
+        required=True,
+        elp='Change all order lines discount.')
+
+    @api.multi
+    def update_discount_lines(self):
+        for sale_obj in self:
+            all_discounts = sale_obj.all_discounts
+            if all_discounts < 100 and all_discounts >= 0:
```

Comment:
In fact,I also don't know why,but the code befor exist wrote like this... and I don't know if it is the need of BR or not.